lib/backend/firestore/firestorebk.go

```diff
@@ -768,36 +768,30 @@ func EnsureIndexes(ctx context.Context, adminSvc *apiv1.FirestoreAdminClient, tu
 		}
 		// operation can be nil if error code is codes.AlreadyExists.
 		if operation != nil {
-			meta, err := operation.Metadata()
-			if err != nil {
-				return trace.Wrap(err)
-			}
-			tuplesToIndexNames[tuple] = meta.Index
+			tasks = append(tasks, operation)
 		}
 	}
 
-	// Instead of polling the Index state, we should wait for the Operation to
-	// finish. Also, there should ideally be a timeout on index creation.
-
-	// check for statuses and block
-	for len(tuplesToIndexNames) != 0 {
-		select {
-		case <-time.After(timeInBetweenIndexCreationStatusChecks):
-		case <-ctx.Done():
-			return trace.ConnectionProblem(ctx.Err(), "context timed out or canceled")
+	for i, operation := range tasks {
+		err := waitOnIndexCreation(ctx, l, operation, tuples[i])
```

Comment:
looks like this counts on `tasks` and `tuples` lining up by index/having the same length, but above you can skip adding the operation to `tasks` if it is nil

Reply:
Fixed by merging operations and tuples into one struct